packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2850,14 +3001,17 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     // because it's 2 single taps, each of which may do different things depending
     // on whether it's a single tap, the first tap of a double tap, the second
     // tap held down, a clean double tap etc.
+    if (_getEffectiveConsecutiveTapCount(details.consecutiveTapCount) == 2) {
```

Comment:
Would it make sense to put the logic in _getEffectiveConsecutiveTapCount on the Details classes directly somehow?  Looking at these calls, I'm thinking that it would be easy for someone to introduce a bug in the future by forgetting about _getEffectiveConsecutiveTapCount and using consecutiveTapCount directly.

However, if it doesn't make sense for the details to be concerned about this kind of stuff, then I agree with how it is now.

Reply:
I can add some comments about this but I think the details classes are platform agnostic at the moment and this particular use of `_getEffectiveConsecutiveTapCount` is only for text selection gestures.

Reply:
Ok that sounds good 👍